addons/godot-next/resources/resource_collections/resource_set.gd

```diff
@@ -15,72 +15,74 @@ class_name ResourceSet
 
 ##### CONSTANTS #####
 
-const DELIMITER = "_?_"
-
-const EMPTY_ENTRY = "[ Empty ]"
 const COLLECTION_NAME = "[ Set ]"
 
 ##### PROPERTIES #####
 
-var _name: = ""
-var _type: Script = null
-
 var _data := {}
 
 ##### NOTIFICATIONS #####
 
+func _init() -> void:
+	resource_name = COLLECTION_NAME
+
+func _get(p_property: String):
+	if p_property.begins_with(PREFIX):
+		return _data.get(p_property.lstrip(PREFIX), null)
+	return null
+
+func _set(p_property: String, p_value) -> bool:
+	if p_property.begins_with(PREFIX):
+		var key = p_property.lstrip(PREFIX)
+		if not p_value:
+			#warning-ignore:return_value_discarded
+			_data.erase(key)
+			property_list_changed_notify()
+			return true
+		elif _data[key].get_script() == p_value.get_script():
+			var res = _instantiate_script(p_value) if p_value is Script else p_value
+			if res:
+				_data[key] = res
+		return true
+	return false
+
 func _get_property_list() -> Array:
-	return [ PropertyInfo.new_dictionary("_data", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_STORAGE).to_dict() ]
+	var list := []
+	if not _type:
+		return list
+	
+	list.append(PropertyInfo.new_group(PREFIX, PREFIX).to_dict())
+	list.append(PropertyInfo.new_dictionary(PREFIX, PROPERTY_HINT_RESOURCE_TYPE, "#%s" % _type.get_path(), PROPERTY_USAGE_DEFAULT).to_dict())
+	if _data.empty():
+		list.append(PropertyInfo.new_nil(PREFIX + EMPTY_ENTRY).to_dict())
+	for item in _data:
```

Comment:
`item` -> `an_item`
These slight style changes are optional, but are just a recommendation.

Reply:
Also, a more descriptive name than 'item' would be apt here. Aren't these typenames or something? Might as well make it `a_typename` if that's the case.

Reply:
changed it ;)